Update Large Datasets documentation with VegaFusion 1.0 information (#2855)
altair-viz/altair: doc/user_guide/large_datasets.rst

VegaFusion
~~~~~~~~~~
`VegaFusion`_ is a third-party package that re-implements most Vega-Lite transforms for evaluation
in the Python kernel.  This makes it possible to scale many Altair charts to millions of rows as long as
they include some form of aggregation.

VegaFusion 1.0 provides two rendering modes that are useful in different situations.

> Mime renderer

Reviewer: ```suggestion
Mime Renderer
```

^^^^^^^^^^^^^

> The `VegaFusion mime renderer`_ automates the preaggregate and filter workflow described above. Right before

Reviewer: How about moving the explanation on what it is useful for to the top of these sections? Seems more important to know what it is for before diving into how it works.

```suggestion
The `VegaFusion mime renderer`_ is a good choice for charts that do not re-aggregate or re-filter data in response to selections. It automates the workflow described in :ref:`preaggregate-and-filter`. Right before
```
Could we link to the correct section? In addition to the suggestion above, you'd need to add the reference before the "Preaggregate and Filter in Pandas" header so that it looks like this

```
.. _preaggregate-and-filter:

Preaggregate and Filter in Pandas
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reviewer: After the first sentence, we could even include a small code snippet just to show how easy it is to use?

```python
import vegafusion as vf
vf.enable()
```

a chart is rendered, VegaFusion extracts the datasets and supported transforms and evaluates them in the
Python kernel. It then removes any unused columns and inlines the transformed data into the chart specification
for rendering. Charts rendered this way are self-contained and do not require the Python kernel or a custom

> notebook extension to display.

Reviewer: Maybe add a last sentence hear that charts produced by the mime renderer can be displayed in all the usual frontends where you'd display an Altair chart? Just to drive home that point.

> The mime renderer is a good choice for charts that do not re-aggregate or re-filter data in response
> to selections.
> 

Reviewer: ```suggestion
```
Remove as added above

> Widget renderer

Reviewer: ```suggestion
Widget Renderer
```

^^^^^^^^^^^^^^^

> The `VegaFusion widget renderer`_ uses a Jupyter Widget extension to maintain a live connection between the displayed
> chart and the Python kernel. This makes it possible for transforms to be evaluated interactively in response to
> changes in selections. Charts rendered this way require a running Python kernel and Jupyter Widget extension to
> display.
> 
> The widget renderer is a good choice for charts that re-aggregate or re-filter data in response to selections.

Reviewer: ```suggestion
The `VegaFusion widget renderer`_ is a good choice for charts that re-aggregate or re-filter data in response to selections. It uses a Jupyter Widget extension to maintain a live connection between the displayed
chart and the Python kernel. This makes it possible for transforms to be evaluated interactively in response to
changes in selections. Charts rendered this way require a running Python kernel and Jupyter Widget extension to display.
```
Same reasoning as above. Moved the explanation on "what it is for" to the start of the section.
